ops.qmd

```diff
@@ -2,84 +2,759 @@
+
 ::: {.callout-tip}
 ## Learning Objectives
 
-* coming soon.
+* Understand what is MLOps and why it is needed
+
+* Learn the architectural patterns for traditional MLOps
+
+* Contrast traditional vs. embedded MLOps across the ML lifecycle 
+
+* Identify key constraints of embedded environments 
+
+* Learn strategies to mitigate embedded ML challenges
+
+* Examine real-world case studies demonstrating embedded MLOps principles
+
+* Appreciate the need for holistic technical and human approaches 
 
 :::
 
 ## Introduction
 
-Explanation: This subsection sets the groundwork for the discussions to follow, elucidating the fundamental concept of MLOps and its critical role in enhancing the efficiency, reliability, and scalability of embedded AI systems. It outlines the unique characteristics of implementing MLOps in an embedded context, emphasizing its significance in the streamlined deployment and management of machine learning models.
+Machine Learning Operations (MLOps), is a systematic approach that combines machine learning (ML), data science, and software engineering to automate the end-to-end ML lifecycle. This includes everything from data preparation and model training to deployment and maintenance. MLOps ensures that ML models are developed, deployed, and maintained efficiently and effectively.
+
+Consider a ridesharing company that wants to deploy a machine-learning model to predict rider demand in real time. The data science team spends months developing a model, but when it's time to deploy, they realize it needs to be compatible with the engineering team's production environment. Deploying the model requires rebuilding it from scratch - costing weeks of additional work. This is where MLOps comes in. 
+
+With MLOps, there are protocols and tools in place to ensure that the model developed by the data science team can be seamlessly deployed and integrated into the production environment. In essence, MLOps removes friction during the development, deployment, and maintenance of ML systems. It improves collaboration between teams through defined workflows and interfaces. MLOps also accelerates iteration speed by enabling continuous delivery for ML models.
+
+For the ridesharing company, implementing MLOps means their demand prediction model can be frequently retrained and deployed based on new incoming data. This keeps the model accurate despite changing rider behavior. MLOps also allows the company to experiment with new modeling techniques since models can be quickly tested and updated.
```

Comment:
Yeah because then you say deployed data here which suggests maybe an edge model, but I think in practice this prediction would be done in the cloud since you want information from many devices

Reply:
Not necessarily, esp. in the context of autonomous vehicles. But you are right that the context is not setup correctly. Will try to improve.